src/main/java/org/opensearch/security/DefaultObjectMapper.java

```diff
@@ -35,21 +35,46 @@
 
 import com.google.common.collect.ImmutableSet;
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.InjectableValues;
 import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
 import com.fasterxml.jackson.databind.exc.InvalidFormatException;
 import com.fasterxml.jackson.databind.exc.MismatchedInputException;
 import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
 import com.fasterxml.jackson.databind.type.TypeFactory;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 
 import org.opensearch.SpecialPermission;
 
+class ConfigMapSerializer extends StdSerializer<Map<String, Object>> {
+    @SuppressWarnings("unchecked")
+    public ConfigMapSerializer() {
+        // Pass Map<String, Object>.class to the superclass
+        super((Class<Map<String, Object>>) (Class<?>) Map.class);
+    }
+
+    @Override
+    public void serialize(Map<String, Object> value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+        gen.writeStartObject();
+        for (Map.Entry<String, Object> entry : value.entrySet()) {
+            if ("password".equals(entry.getKey())) {
```

Comment:
we should convert this into a loop over data-structure to support more than 1 sensitive fields to be redacted. Thoughts?

Reply:
I'll create a Set of sensitive fields. wdyt should be included? `password`, `signing_key` and `exchange_key`? Any others?

Reply:
those should be good!

Reply:
@cwperks What do you think about making this a configuration setting - do you think there is enough of a use case to justify writing up an issue with a proposal?

Reply:
@peternied IMO this list of keys should not be configurable. This list is to obfuscate secrets when calling on the security get config API and should include any key where the value is deemed secret. The Security get config API is called by the security-dashboards-plugin to display an authenticator's configuration and any secret values should not be displayed in plain text.